src/backend/core/url_utils.py

```
@@ -0,0 +1,27 @@
+
+    Strips the 'www.' prefix and normalises to lowercase.
+
+    Examples:
+        "https://www.demoqa.com/elements" → "demoqa.com"
+        "http://the-internet.herokuapp.com" → "the-internet.herokuapp.com"
+        "invalid" → "unknown"
+    """
+    try:
+        parsed = urlparse(url)
+        hostname = parsed.netloc.lower().replace("www.", "")
+        return hostname if hostname else "unknown"
```

Ruff (0.15.7)

[warning] Consider moving this statement to an `else` block

(TRY300)